aptos-move/block-executor/src/view.rs

```diff
@@ -353,7 +432,82 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TAggregator
         state_key: &Self::IdentifierV1,
         _mode: AggregatorReadMode,
     ) -> anyhow::Result<Option<StateValue>> {
-        // TODO: Integrate aggregators.
+        // TODO: Integrate aggregators V1. That is, we can lift the u128 value
+        //       from the state item by passing the right layout here. This can
+        //       be useful for cross-testing the old and the new flows.
+        // self.get_resource_state_value(state_key, Some(&MoveTypeLayout::U128))
         self.get_resource_state_value(state_key, None)
     }
+
+    fn generate_aggregator_v2_id(&self) -> Self::IdentifierV2 {
+        match &self.latest_view {
+            ViewState::Sync(state) => (state.counter.fetch_add(1, Ordering::SeqCst) as u64).into(),
+            ViewState::Unsync(state) => {
+                let mut counter = state.counter.borrow_mut();
+                let id = (*counter as u64).into();
+                *counter += 1;
+                id
+            },
+        }
+    }
+}
+
+// For aggregators V2, values are replaced with identifiers at deserialization time,
+// and are replaced back when the value is serialized. The "lifted" values are cached
+// by the `LatestView` in the aggregators multi-version data structure.
+impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> ValueExchange
+    for LatestView<'a, T, S, X>
+{
+    fn try_exchange(
```

Comment:
Same here. Exchange could mean id --> value and value --> id. It would be nice to use a more explicit name.

Reply:
The name is generic because you can do arbitrary value to value replacement. It's just as using it for IDs. I guess there are no other usages anyway, so renaming might be indeed better👍

Reply:
Actually, looking at that, this is pretty generic, haha, so I think exchange is better. Let me know if you really prefer values/ids